### PR TITLE
Define CPU mode none for virtual, host-model for HW cases

### PR DESCRIPTION
**Pull request by bogdando**
see http://docs.openstack.org/trunk/openstack-compute/install/apt/content/kvm.html

Fixes: #PRD-1950

Signed-off-by: Bogdan Dobrelya bogdando@mail.ru

### CODE REVIEW

File: deployment/puppet/nova/manifests/compute/libvirt.pp [hunk 1]
```diff
@@ -91,6 +91,8 @@
   nova_config {
     'DEFAULT/compute_driver':   value => 'libvirt.LibvirtDriver';
     'DEFAULT/libvirt_type':     value => $libvirt_type;
+    #FIXME use fuel_settings to pass CPU mode: should be none for VM cases (default is host-model)
+    'DEFAULT/libvirt_cpu_mode': value => 'none';
```
Comment (xdel) [inline deployment/puppet/nova/manifests/compute/libvirt.pp]: I`m sorry, but why? Use qemu64 with VMX, seems to be most stable choice.

File: deployment/puppet/nova/manifests/compute/libvirt.pp [hunk 2]
```diff
@@ -91,6 +91,9 @@
   nova_config {
     'DEFAULT/compute_driver':   value => 'libvirt.LibvirtDriver';
     'DEFAULT/libvirt_type':     value => $libvirt_type;
+    #FIXME use fuel_settings to pass CPU mode: none, host-passthrough, host-model and custom
+    #for custom cases, use libvirt_cpu_mode=custom, libvirt_cpu_model=<cpu_model>
+    'DEFAULT/libvirt_cpu_mode': value => 'none';
```
Comment (aglarendil) [inline deployment/puppet/nova/manifests/compute/libvirt.pp]: I think we need to write puppet fact that returns correct value depending on host architecture.

File: deployment/puppet/nova/manifests/compute/libvirt.pp [hunk 3]
```diff
@@ -95,4 +95,14 @@
     'DEFAULT/vncserver_listen': value => $vncserver_listen;
     'DEFAULT/disk_cachemodes': value => '"file=writethrough"';
   }
+
+if $::is_virtual {
```
Comment (aglarendil) [inline deployment/puppet/nova/manifests/compute/libvirt.pp]: change to str2bool($::is_virtual) 'cause facter returns strings

File: deployment/puppet/nova/manifests/compute/libvirt.pp [hunk 4]
```diff
@@ -13,16 +13,16 @@
...
+
 
 #    package { 'qemu':
 #      ensure => present,
 #    }
- 
-    exec { 'symlink-qemu-kvm': 
+
+    exec { 'symlink-qemu-kvm':
       command => "/bin/ln -sf /usr/libexec/qemu-kvm /usr/bin/qemu-system-x86_64",
```
Comment (xdel) [inline deployment/puppet/nova/manifests/compute/libvirt.pp:23]: WTF is that? emulator path can be passed upon vm creation in more proper way

Comment (bogdando) [inline deployment/puppet/nova/manifests/compute/libvirt.pp:23]: I didn't change that. Just whitespaces autofix...

Comment (xdel) [inline deployment/puppet/nova/manifests/compute/libvirt.pp:23]: ok, moving to the seperate pull

**Comment by aglarendil** [comment]
+2

**Pull request merged**